webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.js

```diff
@@ -104,6 +107,15 @@ class BreadcrumbBar extends React.Component {
         {!isTitle && <hr className="breadcrumb-line" />}
       </div>
     );
+
+    if (wrapWithReactRouter) {
+      return (
+        <Router history={history}>
+          <AppSwitcher>{bar}</AppSwitcher>
```

Comment:
This will duplicate all of our switching mechanisms for each breadcrumb, manipulating the history object from different locations would be risky. I think the best option is to keep redux as the one source of truth, even for navigating between pages.
I've encountered with a promising package - [connected-react-router](https://github.com/supasate/connected-react-router) - which binds the router with redux. so for example for navigating between pages we can trigger an action like this:

```js
import { push } from 'connected-react-router'

store.dispatch(push('/path/to/somewhere'))
```
we can create a facory HOC with it so we can wrap it on mounted components

Reply:
I created a PR that uses this package:
https://github.com/theforeman/foreman/pull/7186

works like a charm :)

Reply:
Thanks, I'll use #7189.